rfcs/2021-06-15-shipping-blade.md

```diff
@@ -0,0 +1,355 @@
+
+### Conclusion
+I feel we below is the checklist that we need.
+
+✅ Can use library with the script tag in html(umd bundle).
+
+✅ Library author responsible for tree shaking
+
+❌ Import anything from anywhere
+
+❌ Working of library depends on the build tools config of consumer apps
+
+And this can be achieved by going with the approach of bundling using rollup.
+
+> Not choosing parcel for following reasons:
+> 1. Not familiar with the tool
+> 2. We need to bundle react-native as well so I'm not too sure how smooth it will be with parcel.
+
+## What module system to target?
+
+### CommonJS(a.k.a cjs)
+* Everything that we see using `require` or `module.exports` is commonjs
+* Not great at tree shaking capabilities because when you import you get an object.
+* More code shipped to the browsers which means more download, parse execute times.
+### ES Modules(a.k.a esm)
+* Everything that we see using `import` or `export` is esm.
+* It's great at tree shaking as you can determine imports and exports at compile time (statically) — the tools only have to look at the source code, they don’t have to execute it.
+* Less code shipped to the browsers which means better download, parse execute times.
+
+### Conclusion
+We should definitely going with ESM. There might be few gotchas and unknowns that might pop up but we'll move ahead and fix them instead of taking a back seat and ship CJS.
+
+## Meta information in package.json
+* If we decide to bundle(instead of transpile) then we need to define values for `main`, `module`, `react-native` fields in package.json.
+* If we our final bundle has `components`, `tokens`, `types`, `utils`  as individual pieces then we can't define these values since these fields in `package.json` accepts only single string value. So we might need to explore `exports` field of `package.json`. [Check Preact for reference.](https://github.com/preactjs/preact/blob/master/package.json#L12-L71)
+* if we just produce one final bundle then we can define values for `main`, `module`, `react-native` fields in package.json. I think the above point should be fine.
+
+
+# Drawbacks/Constraints
+I've already spoken about multiple approaches and constraints of each of them in the respective sections itself.
+
+# Alternatives
+I've already spoken about multiple approaches and constraints of each of them in the respective sections itself.
+
+# Adoption strategy
+There's no adoption strategy as such. We'll just need to document the usage of design system once it's published.
+
+# Open Questions
+1. Do we need any changes in our src directory structure to support our needs?
+2. We have custom types defined at a lot of place so how do we merge them with the types generated by tsconfig?
+   * Maybe use something similar to [this library](https://www.npmjs.com/package/npm-dts).
+
+# References
```

Comment:
please add a few references 😄

Reply:
I don't have references for this 🙈 
Because no one talks about it. Let me see if I can add anything

Reply:
Can you add some references around how other popular open-source DS are doing this?